packages/techdocs-common/src/stages/generate/techdocs.ts

```diff
@@ -66,24 +66,36 @@ export class TechdocsGenerator implements GeneratorBase {
   public async run({
     directory,
     dockerClient,
+    expectedResultDir,
     parsedLocationAnnotation,
   }: GeneratorRunOptions): Promise<GeneratorRunResult> {
-    const tmpdirPath = os.tmpdir();
-    // Fixes a problem with macOS returning a path that is a symlink
-    const tmpdirResolvedPath = fs.realpathSync(tmpdirPath);
-    const resultDir = fs.mkdtempSync(
-      path.join(tmpdirResolvedPath, 'techdocs-tmp-'),
-    );
+    // If provided with a directory to output the generated files in, use that directory.
+    // Else create a new temporary directory and return path.
+    let resultDir;
+    if (expectedResultDir) {
+      resultDir = expectedResultDir;
+      await fs.ensureDir(resultDir);
+    } else {
+      const tmpdirPath = os.tmpdir();
+      // Fixes a problem with macOS returning a path that is a symlink
+      const tmpdirResolvedPath = fs.realpathSync(tmpdirPath);
+      resultDir = fs.mkdtempSync(
```

Comment:
bit nitpicky, but essentially we should basically never use `*Sync` functions - we are already in an async function and could use something like `util.promisify(fs.mkdtemp(...))`

Reply:
Thanks! This was from before; replaced with `await fs.mkdtemp()`.